crates/rattler_solve/src/libsolv/wrapper/transaction.rs

```diff
@@ -1,167 +1,70 @@
-                    ffi::SOLVER_TRANSACTION_DOWNGRADED
-                    | ffi::SOLVER_TRANSACTION_UPGRADED
-                    | ffi::SOLVER_TRANSACTION_CHANGED => {
-                        package_operations.push(PackageOperation {
-                            package: repodata_record.clone(),
-                            kind: PackageOperationKind::Remove,
-                        });
-
-                        let solvable_offset =
-                            ffi::transaction_obs_pkg(self.as_ptr().as_ptr(), raw_id);
-                        let second_solvable = SolvableId(solvable_offset);
-                        let second_solvable = second_solvable.resolve(pool);
-                        let second_solvable_index =
-                            second_solvable.get_usize(solvable_index_id).unwrap();
-                        let second_repo_index = repo_mapping[&second_solvable.repo().id()];
-                        let second_repodata_record =
-                            &repodata_records[second_repo_index][second_solvable_index];
-
-                        package_operations.push(PackageOperation {
-                            package: second_repodata_record.clone(),
-                            kind: PackageOperationKind::Install,
-                        });
-                    }
-                    ffi::SOLVER_TRANSACTION_REINSTALLED => {
-                        package_operations.push(PackageOperation {
-                            package: repodata_record.clone(),
-                            kind: PackageOperationKind::Reinstall,
-                        });
-                    }
-                    ffi::SOLVER_TRANSACTION_INSTALL => {
-                        package_operations.push(PackageOperation {
-                            package: repodata_record.clone(),
-                            kind: PackageOperationKind::Install,
-                        });
-                    }
-                    ffi::SOLVER_TRANSACTION_ERASE => {
-                        package_operations.push(PackageOperation {
-                            package: repodata_record.clone(),
-                            kind: PackageOperationKind::Remove,
-                        });
-                    }
-                    ffi::SOLVER_TRANSACTION_IGNORE => {}
-                    _ => {
-                        unsupported_operations.push(id_type);
-                    }
-                }
-            };
+    /// Returns the transaction type
+    pub fn transaction_type(&self, solvable_id: SolvableId) -> ffi::Id {
+        unsafe {
+            ffi::transaction_type(
+                self.raw_ptr(),
+                solvable_id.into(),
+                ffi::SOLVER_TRANSACTION_SHOW_ALL as std::os::raw::c_int,
+            )
         }
+    }
 
-        if !unsupported_operations.is_empty() {
-            return Err(unsupported_operations);
-        }
+    /// Returns the second solvable associated to the transaction
```

Comment:
What does this mean? Can you clarify what `obs` means?

Reply:
I am not entirely sure what obs means here (I took it from mamba). Since this code is being deleted in #75, I suggest we leave it unchanged